Update cargo, clippy stuff

lishaduck opened the pull request:
Depends on #4 (not really, just want to make sure I can run tests)
Here's the diff as GH doesn't play well with stacks from forks: <https://github.com/lishaduck/elm-solve-deps-wasm/compare/update-examples...better-cargo+clippy>

lishaduck committed:
chore: minimal dependency updates

Just what cargo bumps automatically.

lishaduck committed:
fix: typo in comment

lishaduck committed:
chore: remove outdated comment

lishaduck committed:
style: match pedantic style

Good ol' clippy found a thing or two.
This is a high quality codebase—these lints were only added a version or two ago.

lishaduck committed:
chore: add shebangs

For testing.

lishaduck committed:
chore: update lockfile format

NPM whines otherwise.

lishaduck committed:
fix: typos

lishaduck committed:
chore: remove extraneous catch bindings

lishaduck committed:
chore: update metadata

lishaduck committed:
chore: use node protocol

lishaduck committed:
chore: ts

Apparently Prettier and flow comments don't like each other.
Thus, it's a ts project now :)

lishaduck committed:
chore: esm-ify examples

lishaduck committed:
style: prettier

mpizenberg commented:
One question, what made you add a v1.56 minimum rust version?

lishaduck commented:
> One question, what made you add a v1.56 minimum rust version?

You use edition 2021, and that's the version that introduced it.

mpizenberg commented:
Well right, but editions are different than versions. You can compile code that rely on older editions, but not if there is the minimum rust version specified. Edition brings changes on the surface, but the intermediate representation used by the compiler stays the same. This means I can someone compiling a 2018 edition package that depends on this package is totally possible. But if you force Rust version 1.56, and they are compiler their project with rust 1.55, compilation will crash.

Minimum rust version should only be set precisely because some compiler features are needed for the crate to correctly compile. If you didn’t hit a specific issue that led you to add v1.56 is a minimum version, let’s keep it as it is.

lishaduck commented:
> You can compile code that rely on older editions, but not if there is the minimum rust version specified. Edition brings changes on the surface, but the intermediate representation used by the compiler stays the same. This means I can someone compiling a 2018 edition package that depends on this package is totally possible. But if you force Rust version 1.56, and they are compiler their project with rust 1.55, compilation will crash.
> 
> Minimum rust version should only be set precisely because some compiler features are needed for the crate to correctly compile. If you didn’t hit a specific issue that led you to add v1.56 is a minimum version

Ah, that makes sense. I can revert that.
To make sure I understand in the future: you can compile editions from the future on old versions of rust?

lishaduck commented:
Oh, but removing it makes clippy default to using brand new features. I'll take a look in a minute.
Also, crates shouldn't depend on this, they should depend on the Rust one, right?

mpizenberg commented:
Ok, that came out wrong. You are right that since this is using 2021 edition, a compiler older than 1.56 will not be aware of this crate syntax and will fail to compile.

lishaduck commented:
> Ok, that came out wrong. You are right that since this is using 2021 edition, a compiler older than 1.56 will not be aware of this crate syntax and will fail to compile.

So I shouldn't remove the msrv?

lishaduck committed:
chore: bump msrv

Tidy up the code a bit, as there're no Rust consumers of this package.

mpizenberg commented:
I don’t think people set the rust minimum version when there isn’t anything specific that was needed after the edition changes. However, as you some tooling might pick it up as a version you actively support. Let’s keep it blank. And if Clippy wants new changes, let’s do it as it is often right about this stuff.

lishaduck commented:
> I don’t think people set the rust minimum version when there isn’t anything specific that was needed after the edition changes. However, as you some tooling might pick it up as a version you actively support. Let’s keep it blank. And if Clippy wants new changes, let’s do it as it is often right about this stuff.

What I was noting is that clippy defaults to my current rust compiler as the MSRV (v1.79)
So, when I deleted the MSRV, it recommended fixes for Rust v1.58.
Thus, I went back and added that (v1.58) as the MSRV so that it won't fail to compile on older versions, right?

lishaduck commented:
And then, I guess my other thought was the msrv can be bumped whenever, as the only consumers of this are on the JS side, so that bump was fine, right?

lishaduck commented:
Oh, on re-reading your comment, are you saying that you wouldn't set it for this "application," as it's not a crates.io crate, so there's not really a version we "actively support"?
Thus, you'd remove it and only support latest Rust?

lishaduck commented:
> Oh, on re-reading your comment, are you saying that you wouldn't set it for this "application," as it's not a crates.io crate, so there's not really a version we "actively support"? Thus, you'd remove it and only support latest Rust?

I'm going with this interpretation.

lishaduck commented:
Merged in #3.
